jira-feature-collector/src/main/java/com/capitalone/dashboard/client/project/ProjectDataClientImpl.java

```
@@ -45,79 +42,75 @@ public ProjectDataClientImpl(FeatureSettings featureSettings,
-
-				// path - does not exist for Jira
-				scope.setProjectPath(TOOLS.sanitizeResponse(dataMainObj
-						.get("name")));
-
-				try {
+			LOGGER.debug("Size of paged Jira response: ", currentPagedJiraRs.size());
+			if ((currentPagedJiraRs != null) && !(currentPagedJiraRs.isEmpty())) {
+				Iterator<BasicProject> globalResponseItr = currentPagedJiraRs.iterator();
+				while (globalResponseItr.hasNext()) {
+					/*
+					 * Initialize DOMs
+					 */
+					Scope scope = new Scope();
+					BasicProject jiraScope = globalResponseItr.next();
+
+					/*
+					 * Removing any existing entities where they exist in the
+					 * local DB store...
+					 */
+					@SuppressWarnings("unused")
+					boolean deleted = this.removeExistingEntity(TOOLS.sanitizeResponse(jiraScope
+							.getId()));
+
+					/*
+					 * Project Data
+					 */
+					// collectorId
+					scope.setCollectorId(featureCollectorRepository.findByName(Constants.JIRA)
+							.getId());
+
+					// ID;
+					scope.setpId(TOOLS.sanitizeResponse(jiraScope.getId()));
+
+					// name;
+					scope.setName(TOOLS.sanitizeResponse(jiraScope.getName()));
+
+					// beginDate - does not exist for jira
+					scope.setBeginDate("");
+
+					// endDate - does not exist for jira
+					scope.setEndDate("");
+
+					// changeDate - does not exist for jira
+					scope.setChangeDate("");
+
+					// assetState - does not exist for jira
+					scope.setAssetState("Active");
+
+					// isDeleted - does not exist for jira
+					scope.setIsDeleted("False");
+
+					// path - does not exist for Jira
+					scope.setProjectPath(TOOLS.sanitizeResponse(jiraScope.getName()));
+
+					// Saving back to MongoDB
 					projectRepo.save(scope);
-				} catch (Exception e) {
-					LOGGER.error("Unexpected error caused when attempting to save data\nCaused by:\n"
-							+ e.getMessage()
-							+ " : "
-							+ e.getCause()
-							+ "\n"
-							+ Arrays.toString(e.getStackTrace()));
 				}
 			}
 		} catch (Exception e) {
```

Please use specific Exceptions to catch, It is considered a bad practice if you are using Parent Exception class to catch all exception, plus it becomes hard to debug too .